### PR TITLE
Convert variable "rules" to array

### LINKED ISSUE #414: Convert egg variable's rules column array/json cast
This is to help prepare for the yaml egg.

Input:
```
string
required|int
int|min:0|max:12345
required|in:1,3,4,7,11|regex:/asdf/
```

Output:
```json
["string"]
["required", "int"]
["int", "min:0", "max:12345"]
["required", "in:1,3,4,7,11", "regex:/asdf/"]
```

* [ ] Migration to convert existing rows into json
* [ ] Array/JSON cast
* [ ] All usages use new cast

### PR DESCRIPTION
**Pull request by Boy132**
Closes #414

### CODE REVIEW

File: app/Models/EggVariable.php [hunk 1]
```diff
@@ -58,12 +58,14 @@ class EggVariable extends Model
         'default_value' => 'string',
         'user_viewable' => 'boolean',
         'user_editable' => 'boolean',
-        'rules' => 'string',
+        'rules' => 'array',
+        'rules.*' => 'string',
```
Comment (RMartinOscar) [inline app/Models/EggVariable.php:62]:
```suggestion
        'rules.*' => 'string|in:' . implode(',', array_map(fn($rule) => $rule->value, ValidationRule::cases())),
```

File: app/Filament/Resources/EggResource/Pages/CreateEgg.php [hunk 2]
```diff
@@ -173,7 +173,30 @@ public function form(Form $form): Form
...
+                                            'url',
+                                            'email',
+                                            'regex:',
+                                            'min:',
+                                            'max:',
+                                            'between:',
+                                            'between:1024,65535',
+                                            'in:',
+                                            'in:true,false',
+                                        ]),
```
Comment (RMartinOscar) [inline app/Filament/Resources/EggResource/Pages/CreateEgg.php:199]:
```suggestion
                                        ->suggestions(\App\Enums\ValidationRule::cases()),
```

File: app/Filament/Resources/EggResource/Pages/EditEgg.php [hunk 3]
```diff
@@ -183,7 +183,30 @@ public function form(Form $form): Form
...
+                                            'url',
+                                            'email',
+                                            'regex:',
+                                            'min:',
+                                            'max:',
+                                            'between:',
+                                            'between:1024,65535',
+                                            'in:',
+                                            'in:true,false',
+                                        ]),
```
Comment (RMartinOscar) [inline app/Filament/Resources/EggResource/Pages/EditEgg.php:209]:
```suggestion
                                        ->suggestions(\App\Enums\ValidationRule::cases()),
```

**Review by RMartinOscar — COMMENTED** [review]
We should make this an [Enum](https://github.com/pelican-dev/panel/tree/main/app/Enums) as we will use it anytime we want to edit/create an egg variable

Should use the [Port](https://github.com/pelican-dev/panel/blob/main/app/Rules/Port.php) [custom validation rule](https://laravel.com/docs/11.x/validation#custom-validation-rules) instead of using `betwenn:1024,65535`

`app/Enums/ValidationRule.php`
```php
<?php

namespace App\Enums;

use \App\Rules;

enum ValidationRule: string
{
    case Accepted = 'accepted';
    case AcceptedIf = 'accepted_if';
    case ActiveUrl = 'active_url';
    case AfterDate = 'after';
    case AfterOrEqualDate = 'after_or_equal';
    case Alpha = 'alpha';
    case AlphaDash = 'alpha_dash';
    case AlphaNumeric = 'alpha_num';
    case Array = 'array';
    case Ascii = 'ascii';
    case Bail = 'bail';
    case BeforeDate = 'before';
    case BeforeOrEqualDate = 'before_or_equal';
    case Between = 'between';
    case Boolean = 'boolean';
    case Confirmed = 'confirmed';
    case Contains = 'contains';
    case CurrentPassword = 'current_password';
    case Date = 'date';
    case DateEquals = 'date_equals';
    case DateFormat = 'date_format';
    case Decimal = 'decimal';
    case Declined = 'declined';
    case DeclinedIf = 'declined_if';
    case Different = 'different';
    case Digits = 'digits';
    case DigitsBetween = 'digits_between';
    case Dimensions = 'dimensions';
    case Distinct = 'distinct';
    case DoesntStartWith = 'doesnt_start_with';
    case DoesntEndWith = 'doesnt_end_with';
    case Email = 'email';
    case EndsWith = 'ends_with';
    case Enum = 'enum';
    case Exclude = 'exclude';
    case ExcludeIf = 'exclude_if';
    case ExcludeUnless = 'exclude_unless';
    case ExcludeWith = 'exclude_with';
    case ExcludeWithout = 'exclude_without';
    case Exists = 'exists';
    case Extensions = 'extensions';
    case File = 'file';
    case Filled = 'filled';
    case GreaterThan = 'gt';
    case GreaterThanOrEqual = 'gte';
    case HexColor = 'hex';
    case Image = 'image';
    case In = 'in';
    case InArray = 'in_array';
    case Integer = 'integer';
    case IPAddress = 'ip';
    case JSON = 'json';
    case LessThan = 'lt';
    case LessThanOrEqual = 'lte';
    case List = 'list';
    case Lowercase = 'lowercase';
    case MACAddress = 'mac_address';
    case Max = 'max';
    case MaxDigits = 'max_digits';
    case MIMETypes = 'mimes';
    case MIMETypeByFileExtension = 'mimetypes';
    case Min = 'min';
    case MinDigits = 'min_digits';
    case Missing = 'missing';
    case MissingIf = 'missing_if';
    case MissingUnless = 'missing_unless';
    case MissingWith = 'missing_with';
    case MissingWithAll = 'missing_with_all';
    case MultipleOf = 'multiple_of';
    case NotIn = 'not_in';
    case NotRegex = 'not_regex';
    case Nullable = 'nullable';
    case Numeric = 'numeric';
    case Present = 'present';
    case PresentIf = 'present_if';
    case PresentUnless = 'present_unless';
    case PresentWith = 'present_with';
    case PresentWithAll = 'present_with_all';
    case Prohibited = 'prohibited';
    case ProhibitedIf = 'prohibited_if';
    case ProhibitedUnless = 'prohibited_unless';
    case Prohibits = 'prohibits';
    case RegularExpression = 'regex';
    case Required = 'required';
    case RequiredIf = 'required_if';
    case RequiredIfAccepted = 'required_if_accepted';
    case RequiredIfDeclined = 'required_if_declined';
    case RequiredUnless = 'required_unless';
    case RequiredWith = 'required_with';
    case RequiredWithAll = 'required_with_all';
    case RequiredWithout = 'required_without';
    case RequiredWithoutAll = 'required_without_all';
    case RequiredArrayKeys = 'required_array_keys';
    case Same = 'same';
    case Size = 'size';
    case Sometimes = 'sometimes';
    case StartsWith = 'starts_with';
    case String = 'string';
    case Timezone = 'timezone';
    case Unique = 'unique';
    case Uppercase = 'uppercase';
    case URL = 'url';
    case ULID = 'ulid';
    case UUID = 'uuid';
    // Customs
    case Port = new Port;
    case Username = new Username;
}
```

**Comment by Boy132** [comment]
> We should make this an [Enum](https://github.com/pelican-dev/panel/tree/main/app/Enums) as we will use it anytime we want to edit/create an egg variable

This is just overly complicated. Unless Laravel provides a first party enum with all available rules we shouldn't add this.
I only added the most used rules to suggestions to make it easier but didn't intend it to act as full auto-complete.

**Comment by RMartinOscar** [comment]
> > We should make this an [Enum](https://github.com/pelican-dev/panel/tree/main/app/Enums) as we will use it anytime we want to edit/create an egg variable
> 
> 
> 
> This is just overly complicated. Unless Laravel provides a first party enum with all available rules we shouldn't add this.
> 
> I only added the most used rules to suggestions to make it easier but didn't intend it to act as full auto-complete.

I thought it was an auto-complete mybad and Laravel doesn't seem to have an Enum of all the default rules

**Review by notAreYouScared — APPROVED** [review]
LGTM, rules appear to still work for validation on both admin and client sides.

**Pull request merged**